Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +212,49 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
+        $document = $this->dm->find(null, $path);
 
-        return $this->dm->find(null, $path);
+        if ($document instanceof AutoRouteInterface) {
+            return $document;
+        }
+
+        return null;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * Convert the given generic document to an auto route document, migrating any children which it may have.
+     *
+     * @param Generic $document
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
```

Comment:
How does it fail? @lsmith77 I would expect that solution to throw an Exception because the Generic document would not be an instance of `$autoRouteClassName` or is `Generic` (unmanaged) a special case? And if so, and that doesn't work, perhaps we should indeed handle it somehow.

Otherwise I am fine with the code here if there is no existing other solution.